### DISCUSSION TITLE
Firewall mode only

### DISCUSSION THREAD

**Discussion by AdamJel** [celzero/rethink-app#124] [Ideas]
Currently it is not possible to start the service without dns in firewall mode only. I use a paid vpn provider and trust him alone. So I would like to use only the firewall alone, which is not possible. To start the service, it is needed to set always-on-vpn setting in android to off, to disconnect from current vpn and to connect through rethinkndns. 

It would be nice, if it was possible to use the firewall feature only and remain the other vpn service.

**Comment by ignoramous** [comment]
Thanks for your feedback. You've flagged multiple unrelated issues here. Here's the breakdown:

#### Use third-party VPN and Rethink Firewall at the same time

This isn't possible on Android since chaining VPNs isn't allowed. But:

1. v05x supports forwarding connections to any SOCKS5 endpoint (like to Tor via Orbot Android app, for example). To my knowledge, NordVPN supports SOCKS5. Not sure about other providers. Do note that SOCKS5 doesn't encrypt traffic but changes IP address. #100 

2. We'd add support for configuring the application with WireGuard in next two months or so: #52 Most popular VPNs including Mullvad support WireGuard now. What this means is that you'd be able to forward all of the traffic from Rethink to any WireGuard endpoint. We may add support for OpenVPN, though don't hold your breath on that: #64 

3. We will add support for Firewall on rooted devices that'd help run VPN side by side with it and probably be infinitely more efficient on the battery at the same time (and infinitely more powerful). This is long time away. #66 and #78 

#### Always-on VPN and Block connections without VPN support

We've fixed that in the latest builds (not released to public yet) #92
